packages/components/button/src/ButtonGroup/ButtonGroup.tsx

```diff
@@ -39,17 +39,24 @@ function _ButtonGroup(
       ref={ref}
       className={cx(styles.buttonGroup, className)}
     >
-      {React.Children.map(children, (child, key) => {
+      {React.Children.map(children, (child, index) => {
         if (!child) {
           return null;
         }
-        return React.cloneElement(child as React.ReactElement, {
-          key,
-          className: cx(
-            styles.groupContent,
-            (child as React.ReactElement).props.className,
-          ),
-        });
+        return (
+          <>
+            {withDivider && index !== 0 ? (
+              <div className={styles.divider} />
+            ) : null}
```

Comment:
🧹 I suggest using the `&&` notion instead of a null return, but thats just a nitpick as i think rendering an additional html tag for every element is the bigger issue here.